### DISCUSSION TITLE
Expose captureScreenshot as a beaker API

### DISCUSSION THREAD

**Discussion by willurmston** [beakerbrowser/beaker#1741] [General] [answered]
It would be great to be able to use the built-in `captureScreenshot` function as an API. This is something you would traditionally need to set up a server or use an existing API for, and isn't possible in the browser due to CORS restrictions.

I'm imagining something like...
```
await beaker.screenshot.capture( url, { width, height, timeout })
```
Maybe there are horrible security issues with this, and maybe you'd want people to accomplish this with Beaker node, but it would open up some cool uses around embedding, social bookmarking, or visual regression testing (just some ideas).

**Comment by pfrazee** [comment, marked as answer]
Yeah I'm thinking about the right way to approach this, will keep you updated
